### PR TITLE
[release-4.5] Bug 1880324: fix bug in reflector not recovering from "Too large reso…

### PR DESCRIPTION
**Pull request by soltysh**
…urce version"

/assign @p0lyn0mial

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,20 +6,20 @@ require (
...
-	k8s.io/apimachinery v0.18.3
-	k8s.io/apiserver v0.18.3
-	k8s.io/client-go v0.18.3
-	k8s.io/component-base v0.18.3
+	k8s.io/api v0.18.9
+	k8s.io/apimachinery v0.18.9
+	k8s.io/apiserver v0.18.9
+	k8s.io/client-go v0.18.9
+	k8s.io/component-base v0.18.9
 	k8s.io/klog v1.0.0
```
Comment (p0lyn0mial) [inline go.mod:23]: make it v2.0.0

File: vendor/k8s.io/client-go/tools/cache/reflector.go [hunk 2]
```diff
@@ -256,13 +256,14 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 
 			list, paginatedResult, err = pager.List(context.Background(), options)
-			if isExpiredError(err) {
-				r.setIsLastSyncResourceVersionExpired(true)
-				// Retry immediately if the resource version used to list is expired.
+			if isExpiredError(err) || isTooLargeResourceVersionError(err) {
```
Comment (p0lyn0mial) [inline vendor/k8s.io/client-go/tools/cache/reflector.go:259]: this is what we want

File: go.mod (hunk 1 above)
Comment (p0lyn0mial) [reply go.mod:23]: ah, sorry this is 4.5

Comment (p0lyn0mial) [reply go.mod:23]: v1.0.0 is okay - https://github.com/kubernetes/kubernetes/blob/release-1.18/go.mod#L155

Comment (soltysh) [reply go.mod:23]: :wink:

**Comment by eparis** [comment]
pulling cp-approved. did not merge in time this week. should re-evaluate next week.

**Pull request merged**